### PR TITLE
[NN] Add TAGCN nn.module and example

### PR DESCRIPTION
**Pull request by classicsong**
## Description
Add TAGCN nn.module on dgl.nn.pytorch
Add TAGCN example under example.pytorch
Add unitest for TAGCN

#756 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

File: examples/pytorch/tagcn/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+Results
+-------
+Run with following (available dataset: "cora", "citeseer", "pubmed")
+```bash
+python3 train.py --dataset cora --gpu 0 --self-loop
+```
+
+* cora: ~0.812 (0.804-0.823) (paper: 0.833)
+* citeseer: ~0.715 (paper: 0.714)
+* pubmed: ～0.790 (paper: 0.811)
```
Comment (yzh119) [inline examples/pytorch/tagcn/README.md]: Try tuning some hyper-parameters like weight_decay and see if the result is still lower then the number in TAGCN paper.

Comment (classicsong) [reply examples/pytorch/tagcn/README.md]: I tried weight_decay and lr for pubmed dataset, there is minor performance improvement.
The new test result was updated.

File: examples/pytorch/tagcn/train.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl import DGLGraph
+from dgl.data import register_data_args, load_data
+
+from tagcn import TAGCN
+
+#from gcn import GCN
```
Comment (yzh119) [inline examples/pytorch/tagcn/train.py]: Please remove these comments

File: python/dgl/nn/pytorch/conv.py [hunk 3]
```diff
@@ -150,6 +150,95 @@ def extra_repr(self):
...
+        Number of output features.
+    K: int, optional
+        Number of hops :math: `K`. (default: 3)
+    bias: bool, optional
+        If True, adds a learnable bias to the output. Default: ``True``.
+    activation: callable activation function/layer or None, optional
+        If not None, applies an activation function to the updated node features.
+        Default: ``None``.
+
+    Attributes
```
Comment (yzh119) [inline python/dgl/nn/pytorch/conv.py]: Describe attributes here.

File: python/dgl/nn/pytorch/conv.py [hunk 4]
```diff
@@ -150,6 +150,95 @@ def extra_repr(self):
...
+                 in_feats,
+                 out_feats,
+                 k=2,
+                 bias=True,
+                 activation=None):
+        super(TGConv, self).__init__()
+        self._in_feats = in_feats
+        self._out_feats = out_feats
+        self._k = k
+
```
Comment (yzh119) [inline python/dgl/nn/pytorch/conv.py]: remove unnecessary empty lines.

**Comment by jermainewang** [comment]
Since Cora/Citeseer/Pubmed are so common and their training loop are essentially the same, is it possible to put these modules in one example?

**Comment by yzh119** [comment]
Yes we can create a "citation network" folder in examples.

**Comment by yzh119** [comment]
Also, I recommend change the name of this clas from `TAGCN` to something ended with `Conv`, to be consistent with other class in this module.

**Comment by classicsong** [comment]
> Also, I recommend change the name of this clas from `TAGCN` to something ended with `Conv`, to be consistent with other class in this module.

Actually, in nn/pytorch/conv.py it is called TGConv.

**Comment by classicsong** [comment]
> Since Cora/Citeseer/Pubmed are so common and their training loop are essentially the same, is it possible to put these modules in one example?

I don't think so. These modules/examples can also be applied to other datasets. They are not bounded only with these three datasets actually.

File: python/dgl/nn/pytorch/conv.py [hunk 5]
```diff
@@ -150,6 +150,95 @@ def extra_repr(self):
...
+    where :math:`\mathbf{A}` denotes the adjacency matrix and
+    :math:`D_{ii} = \sum_{j=0} A_{ij}` its diagonal degree matrix.
+
+    Parameters
+    ----------
+    in_feats : int
+        Number of input features.
+    out_feats : int
+        Number of output features.
+    K: int, optional
```
Comment (yzh119) [inline python/dgl/nn/pytorch/conv.py]: lowercase

**Pull request merged**